feat: use nix for image/package build and remove rust (#951)
fkie-cad/Logprep: doc/source/installation.rst

Installation
============

PyPI
====

This option is recommended for using logprep as a library to build upon

.. code-block:: bash

    pip install logprep

To see if the installation was successful run :code:`logprep --version`.

Container Image
===============

This option is recommended for Users just wanting to run logprep without the hassle of compiling.
We provide a number of pre compiled Images for all officially supported python versions.

> A list of every image can be found here https://github.com/fkie-cad/Logprep/pkgs/container/logprep/versions?filters%5Bversion_type%5D=tagged.

Reviewer: ```suggestion
This option is recommended for users wanting to run logprep without the hassle of compiling.
We provide precompiled images for all python versions supported by logprep.
A list of available images can be found here https://github.com/fkie-cad/Logprep/pkgs/container/logprep/versions?filters%5Bversion_type%5D=tagged.
```

To install a specific version the general structure of our tags follows this example:

logprep:py{PYTHON_VERSION}-v{LOGPREP_VERSION}

> So python 3.11 with version 19.0.0 would be:

Reviewer: ```suggestion
Python 3.11 with version 19.0.0 would be called:
```

logprep:py3.11-v19.0.0

> To download (pull) these Images an OCI compliant Puller (in this example Docker) can be used like this

Reviewer: ```suggestion
To download (pull) these images an OCI compliant puller (like docker) can be used like this
```

    uv sync --frozen
    uv sync --frozen --extra dev # if you intend to contribute

There is also the option of using nix flakes.

Flakes are theoretically experimental so depending on which nix installer you used,
you might have to follow this short guide:

https://wiki.nixos.org/wiki/Flakes#Nix_standalone

> after that, you should be able to run the following code and just start developing.

Reviewer: ```suggestion
After that, you should be able to run the following code and just start developing.
```
